common/types/offer.go

```diff
@@ -2,60 +2,35 @@ package types
 
 import (
 	"crypto/rand"
-	"encoding/hex"
-	"encoding/json"
 	"fmt"
 
-	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/Masterminds/semver/v3"
 	"golang.org/x/crypto/sha3"
 )
 
-// Hash represents a 32-byte hash
-type Hash [32]byte
-
-// EmptyHash is an empty Hash
-var EmptyHash = Hash{}
-
-// String returns the hex-encoded hash
-func (h Hash) String() string {
-	return hex.EncodeToString(h[:])
-}
-
-// IsZero returns true if the hash is all zeros, otherwise false
-func (h Hash) IsZero() bool {
-	return h == [32]byte{}
-}
-
-// HexToHash decodes a hex-encoded string into a hash
-func HexToHash(s string) (Hash, error) {
-	h, err := hex.DecodeString(s)
-	if err != nil {
-		return [32]byte{}, err
-	}
-
-	var hash [32]byte
-	copy(hash[:], h)
-	return hash, nil
-}
+// CurOfferVersion is the latest supported version of a serialised Offer struct
+var CurOfferVersion, _ = semver.NewVersion("v0.1.0")
 
 // Offer represents a swap offer
 type Offer struct {
-	id            Hash
-	Provides      ProvidesCoin
-	MinimumAmount float64
-	MaximumAmount float64
-	ExchangeRate  ExchangeRate
-	EthAsset      EthAsset
+	Version       *semver.Version `json:"version"`
+	ID            Hash            `json:"offer_id"`
+	Provides      ProvidesCoin    `json:"provides"`
+	MinimumAmount float64         `json:"min_amount"`
+	MaximumAmount float64         `json:"max_amount"`
```

Comment:
I'd been wanting to shorten these variables. With the JSON decoupled from the variable names, we can still do it at some point.  :)